packages/zk-certificates/lib/queryMerkleTree.ts

```diff
@@ -1,7 +1,9 @@
 /* Copyright (C) 2023 Galactica Network. This file is part of zkKYC. zkKYC is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. zkKYC is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with this program. If not, see <https://www.gnu.org/licenses/>. */
-import type { HardhatEthersHelpers } from '@nomiclabs/hardhat-ethers/types';
+import { buildPoseidon } from 'circomlibjs';
+import type { Contract, providers } from 'ethers';
 
 import { printProgress } from './helpers';
```

Comment:
This function won't work in a browser.
we can pass callback in this function
something like this:
```
type Args = {
  provider: providers.Provider;
  registry: KYCRecordRegistry;
  firstBlock?: number;
  onProgress?: (percent: string) => void;
};
```

Reply:
Good point, I'll change it.